File: 2018-12-29/1475/1475.c
#include <stdio.h>
#include <string.h>

int numberSet[11] = {0,};

int setSetNumber(int length, char N[]);
int getMaxSet();

int main(){
	char N[9];
	scanf("%s", &N);
	int Nlength = strlen(N);
	printf("%d", setSetNumber(Nlength, N));

}

int setSetNumber(int length, char N[]){
	for(int i = 0; i < length; i++)
		numberSet[N[i]-'0']+=1;
	int result = getMaxSet();
	return result;
}

int getMaxSet(){
	int sumSixNine = numberSet[6] + numberSet[9];
	int max = 0;

	numberSet[6] = sumSixNine/2 + sumSixNine%2;

	for(int i = 0; i < 9; i++){
		if(max < numberSet[i])
			 max = numberSet[i];
	}
	return max;
}

File: 2018-12-29/1475/1475.java
import java.io.*;

class Main{
	public static void main(String[] args) throws Exception{
		BufferedReader bfr = new BufferedReader(new InputStreamReader(System.in));
		String N = bfr.readLine();
		int result = setSetNumber(N);
		System.out.println(result);
	}
	public static int setSetNumber(String N){
		int [] usingNumber = new int[10];
		for(int i = 0; i < NLength(N); i++)
			usingNumber[toNumber(N.charAt(i))]+=1;

		return getMax(usingNumber);
	}

	public static int NLength(String N){
		return N.length();
	}

	public static int toNumber(char number){
		return number - '0';
	}	

	public static int getMax(int[] usingNumber){
		int max = 0;
		int nineSix = uniteSixNine(usingNumber);
		usingNumber[6] = nineSix;

		for(int i = 0; i < 9; i++)
			if(max < usingNumber[i])
				max = usingNumber[i];
		return max;	
	}

	public static int  uniteSixNine(int[] usingNumber){
		int nineSix = usingNumber[6] + usingNumber[9];
		return nineSix/2 + nineSix%2;
	}
	
}